wellcomeml/ml/bert_classifier.py

```diff
@@ -213,22 +213,22 @@ def partial_fit(self, X, Y, classes=None):
         return self
 
     def predict(self, X):
-        def binarize_output(x):
-            cats = self.nlp(x).cats
+        def binarize_output(doc):
+            cats = doc.cats
             out = [
                 1 if cats[label] > self.threshold else 0 for label in self.unique_labels
             ]
             return out
-
-        return np.array([binarize_output(x) for x in X])
+        doc_gen = self.nlp.pipe(X)
```

Comment:
just looking at Inez's examples, are you sure you dont have to `list()` this?

Reply:
if you want to transform from a generator to a list you can use list but both work in a for loop

Reply:
i named it generator to remind myself that is a genator because when i was testing the speed impact it seems that it was going really fast but it was only because a genetator was running the predictions only when called. anyway details.